pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,219 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/operator-framework/operator-sdk/pkg/metrics"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	k8sclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine/nodeconfig"
+)
+
+var (
+	log = logf.Log.WithName("metrics")
+	// metricsEnabled specifies if metrics are enabled in the current cluster
+	metricsEnabled = true
+)
+
+const (
+	// windowsMetricsEndpoints is the name of the Endpoints object for Windows metrics
+	windowsMetricsEndpoints = "windows-machine-config-operator-metrics"
+	// metricsPortName specifies the portname used for prometheus monitoring
+	PortName = "metrics"
+	// Host is the host address used by Windows metrics
+	Host = "0.0.0.0"
+	// Port is the port number on which windows-exporter is exposed.
+	Port int32 = 9182
+)
+
+// prometheusNodeConfig holds the information required to configure prometheus, so that it can scrape metrics from the
+// given endpoint address
+type prometheusNodeConfig struct {
+	// k8sclientset holds the information related to kubernetes clientset
+	k8sclientset *kubernetes.Clientset
+	// namespace is the namespace in which metrics endpoints object is created
+	namespace string
+}
+
+// patchEndpoint contains information regarding patching metrics Endpoint
+type patchEndpoint struct {
+	// op defines patch operation to be performed on the Endpoints object
+	Op string `json:"op"`
+	// path defines the location of the patch
+	Path string `json:"path"`
+	// value defines the data to be patched
+	Value []v1.EndpointAddress `json:"value"`
+}
+
+// NewPrometheuopsNodeConfig creates a new instance for prometheusNodeConfig  to be used by the caller.
+func NewPrometheusNodeConfig(clientset *kubernetes.Clientset) (*prometheusNodeConfig, error) {
+	windowsMetricsEndpointsNamespace, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, err
+	}
+
+	return &prometheusNodeConfig{
+		k8sclientset: clientset,
+		namespace:    windowsMetricsEndpointsNamespace,
+	}, err
+}
+
+// Add will create the Services and Service Monitors to allow the operator export the metrics by using
+// the Prometheus operator
```

Comment:
`Add will create the Services and Service Monitors that allows the operator to export the metrics...`